snuba/web/views.py

```diff
@@ -364,12 +366,46 @@ def sdk_distribution(*, timer: Timer) -> Response:
     return format_result(run_query(dataset, request, timer))
 
 
+SUBSCRIPTION_SEPARATOR = "/"
+
+
+@application.errorhandler(InvalidSubscriptionError)
+def handle_subscription_error(exception: InvalidSubscriptionError):
+    data = {"error": {"type": "subscription", "message": str(exception)}}
+    return (
+        json.dumps(data, indent=4),
+        400,
+        {"Content-Type": "application/json"},
+    )
+
+
+def build_external_subscription_id(partition_id: str, subscription_id: str) -> str:
+    return "{}{}{}".format(partition_id, SUBSCRIPTION_SEPARATOR, subscription_id)
+
+
 @application.route("/<dataset:dataset>/subscriptions", methods=["POST"])
-def create_subscription(*, dataset: Dataset):
-    partition = 0
-    key = uuid1().hex
+@util.time_request("subscription")
+def create_subscription(*, dataset: Dataset, timer: Timer):
+    data = parse_request_body(http_request)
+    # TODO: Add json schema and validate
+    # TODO: Check for valid queries with fields that are invalid for subscriptions.  For
+    # example date fields and aggregates.
+    partition_id, subscription = SubscriptionCreator(dataset).create(
+        data["project_id"],
+        data["conditions"],
+        data["aggregations"],
+        timedelta(minutes=data["time_window"]),
+        timedelta(minutes=data["resolution"]),
+        timer,
+    )
     return (
-        json.dumps({"subscription_id": f"{partition}/{key}"}),
+        json.dumps(
+            {
+                "subscription_id": build_external_subscription_id(
+                    partition_id, subscription.id
+                )
```

Comment:
I think that `(partition_id, subscription.id)` should be a standalone type (probably a namedtuple) and also the return value of `create`.